swift/ITSClient/Sources/ITSMobility/RegionOfInterest/RegionOfInterestCoordinator.swift

```diff
@@ -0,0 +1,110 @@
+    func updateRoadPositionRegionOfInterest(
+        latitude: Double,
+        longitude: Double,
+        zoomLevel: Int,
+        namespace: String
+    ) -> TopicUpdateRequest? {
+        updateRegionOfInterest(for: .cam,
+                               latitude: latitude,
+                               longitude: longitude,
+                               zoomLevel: zoomLevel,
+                               namespace: namespace,
+                               currentRegionOfInterest: &currentCAMRegionOfInterest)
+    }
+
+    private func updateRegionOfInterest(
+        for messageType: MessageType,
+        latitude: Double,
+        longitude: Double,
+        zoomLevel: Int,
+        namespace: String,
+        currentRegionOfInterest: inout RegionOfInterest?) -> TopicUpdateRequest? {
+        let quadkey = quadkeyBuilder.quadkeyFrom(latitude: latitude,
+                                                 longitude: longitude,
+                                                 zoomLevel: zoomLevel,
+                                                 separator: "/")
+
+        guard quadkey != currentRegionOfInterest?.quadkey else {
+            return nil
+        }
+
+        let neighborQuadkeys = quadkeyBuilder.neighborQuadkeys(for: quadkey)
+        let regionOfInterest = RegionOfInterest(quadkey: quadkey,
+                                                neighborQuadkeys: neighborQuadkeys)
+        let topicUpdate = updateTopicSubscriptions(newRegionOfInterest: regionOfInterest,
+                                                   currentRegionOfInterest: currentRegionOfInterest,
+                                                   for: messageType,
+                                                   namespace: namespace)
+        currentRegionOfInterest = regionOfInterest
+
+        return topicUpdate
+    }
+
+    private func updateTopicSubscriptions(
+        newRegionOfInterest: RegionOfInterest,
+        currentRegionOfInterest: RegionOfInterest?,
+        for messageType: MessageType,
+        namespace: String
+    ) -> TopicUpdateRequest {
+        var subscriptions = [String]()
+        var unsubscriptions = [String]()
+
+        let newQuadkeys = newRegionOfInterest.allQuadkeys
+        let currentQuadkeys = currentRegionOfInterest?.allQuadkeys ?? []
+
+        newQuadkeys.forEach { newQuadkey in
+            if !currentQuadkeys.contains(newQuadkey) {
+                subscriptions.append(topic(for: messageType, in: newQuadkey, namespace: namespace))
+            }
+        }
+
+        currentQuadkeys.forEach { currentQuadkey in
+            if !newQuadkeys.contains(currentQuadkey) {
+                unsubscriptions.append(topic(for: messageType, in: currentQuadkey, namespace: namespace))
+            }
+        }
+
+        return TopicUpdateRequest(subscriptions: subscriptions, unsubscriptions: unsubscriptions)
+    }
+
+    private func topic(for messageType: MessageType, in quadkey: String, namespace: String) -> String {
```

Comment:
What will happen if _quadkey_ or _namespace_ aare empty or not well formatted? Is it even possible at this level?